apollo-mockserver/src/commonMain/kotlin/com/apollographql/apollo3/mockserver/MockServerCommon.kt

```diff
@@ -27,70 +29,50 @@ class MockRequest(
 
 suspend fun writeResponse(sink: BufferedSink, mockResponse: MockResponse, version: String) {
   sink.writeUtf8("$version ${mockResponse.statusCode}\r\n")
-  val isChunked = mockResponse.chunks.isNotEmpty()
-
-  val headers = mockResponse.headers +
-      if (isChunked) {
-        mapOf("Transfer-Encoding" to "chunked")
-      } else {
-        mapOf("Content-Length" to mockResponse.body.size.toString())
-      } +
-      // We don't support 'Connection: Keep-Alive', so indicate it to the client
-      mapOf("Connection" to "close")
-
+  // We don't support 'Connection: Keep-Alive', so indicate it to the client
+  val headers = mockResponse.headers + mapOf("Connection" to "close")
   headers.forEach {
     sink.writeUtf8("${it.key}: ${it.value}\r\n")
   }
   sink.writeUtf8("\r\n")
   sink.flush()
 
-  if (isChunked) {
-    // Chunked format is a sequence of:
-    // - chunk-size (in hexadecimal) + CRLF
-    // - chunk-data + CRLF
-    // Ended with a chunk-size of 0 + CRLF + CRLF
-    for (chunk in mockResponse.chunks) {
-      delay(chunk.delayMillis)
-      sink.writeHexadecimalUnsignedLong(chunk.body.size.toLong())
-      sink.writeUtf8("\r\n")
-      sink.write(chunk.body)
-      sink.writeUtf8("\r\n")
-      sink.flush()
-    }
-    sink.writeUtf8("0\r\n\r\n")
-    sink.flush()
-  } else if (mockResponse.body.size > 0) {
-    sink.write(mockResponse.body)
+  mockResponse.body.collect {
+    sink.write(it)
     sink.flush()
   }
 }
 
 class MockResponse(
     val statusCode: Int = 200,
-    val body: ByteString = ByteString.EMPTY,
-    val chunks: List<MockChunk> = emptyList(),
-    val headers: Map<String, String> = emptyMap(),
+    val body: Flow<ByteString> = emptyFlow(),
+    val headers: Map<String, String> = mapOf("Content-Length" to "0"),
     val delayMillis: Long = 0,
```

Comment:
Can we remove this as a property and "just" delay in the `Flow`?

Reply:
We could but the semantics of this delay are actually a bit different, as it applies before *anything* is sent. Whereas if we did this, the server would send the status code and headers immediately, then delay before sending the body. Maybe that's ok though?

Reply:
Aaahh yes, good catch 👍